File: csharp/static-site/src/StaticSite/StaticSiteStack.cs
using Amazon.CDK;

namespace StaticSite
{
    public class StaticSiteStack : Stack
    {
        //
        // This stack relies on getting the domain name from CDK context.
        // Use 'cdk synth -c domain=mystaticsite.com -c subdomain=www'
        // Or add the following to cdk.context.json:
        // {
        //   "context": {
        //     "domain": "mystaticsite.com",
        //     "subdomain": "www"
        //   }
        // }
        // 
        public StaticSiteStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            // You can also define your own constructs and use them in your stack.
            var hello = new StaticSiteConstruct(this, "MyStaticSite", new StaticSiteConstructProps
            {
                DomainName= (string) this.Node.TryGetContext("domain"),
                SiteSubDomain= (string) this.Node.TryGetContext("subdomain")
            });
        }
    }
}

File: python/static-site/static_site/static_site_stack.py
from aws_cdk import (
    core
)

from .static_site_construct import StaticSiteConstruct, StaticSiteProps


class MyStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        props: StaticSiteProps = StaticSiteProps(
            domain_name=scope.node.try_get_context('domain'),
            site_sub_domain=scope.node.try_get_context('sub_domain'),
            hosted_zone_id=scope.node.try_get_context('hosted_zone_id')
        )

        print(props.domain_name)
        print(props.site_sub_domain)

        StaticSiteConstruct(self, "MyStaticSiteConstruct", props)
